pkg/utils/gardener/dns.go

```diff
@@ -51,10 +54,10 @@ const (
 	InternalDomainKey = "internal"
 )
 
-// GetDomainInfoFromAnnotations returns the provider and the domain that is specified in the given annotations.
-func GetDomainInfoFromAnnotations(annotations map[string]string) (provider string, domain string, includeZones, excludeZones []string, err error) {
+// GetDomainInfoFromAnnotations returns the provider, domain, and zones that are specified in the given annotations.
+func GetDomainInfoFromAnnotations(annotations map[string]string) (provider string, domain string, zone string, includeZones, excludeZones []string, err error) {
```

Comment:
6-Tuple as return value is error-prone. Wouldn't it be clearer to return a struct with the fields provider,domain,zone,includeZones,excludeZones ?

Reply:
Well, I agree, but I am adapting existing code. 6-tuple is not more error-prone than 5-tuple, and this 5-tuple is used in a dozen places at least. I would keep it as it is in this PR and perhaps refactor it in a separate PR.